latex.bzl

```diff
@@ -12,14 +12,120 @@ LatexOutputInfo = provider(
+    Also kpathsea tries to be efficient about looking up files.
+    So to derive if a folder is of interest, kpathsea checks if the number
+    of files or folders in the current folder being inspected is greater
+    than 2.
+    Unfortunately symlinks are (currently) not counted. And Bazel makes heavy
+    use of symlinks.
+
+    However, kpathsea makes heavy use of environment variables (and
+    ls-R database, IIRC).
+    So we can work around this limitation. Also, by only adding the
+    search paths of the files mapped to the environment variable we can reduce
+    the search space, and reduce build times.
+    https://tug.org/texinfohtml/kpathsea.html#Supported-file-formats,
+    lists all environment variables one can set.
+
+    Args:
+      ctx: For accessing the inputs parameters.
+      toolchain: The latex toolchain.
+      files: all files that might be needed as part of the build.
+
+    Returns:
+      A list of commands to provide to the ctx.actions.run invokation
+    """
+
+    def list_unique_folders_from_file_ext(files, exts):
+        directories = []
+        for inp in files:
+            dirname = inp.dirname
+            valid = False
+            if not exts:
+                valid = True
+            else:
+                for ext in exts:
+                    if inp.path.endswith(ext):
+                        valid = True
+            if valid and dirname not in directories:
+                directories.append(dirname)
+        return directories
+
+    def setup_env_for(
+            type_env_dict,
+            env_var,
+            files,
+            extensions = [],
+            post_additions = ""):
+        search_folders = list_unique_folders_from_file_ext(
+            files,
+            extensions,
+        )
+        type_env_dict[env_var] = ".:{}{}".format(
+            ":".join(search_folders),
+            post_additions,
+        )
+
+    type_env = {}
+    setup_env_for(type_env, "AFMFONTS", files, [".afm"])
+    setup_env_for(type_env, "BIBINPUTS", files, [".bib"])
+    setup_env_for(type_env, "ENCFONTS", files, [".enc"])
+    setup_env_for(
+        type_env,
+        "LUAINPUTS",
+        files,
+        [".lua" or ".luc"],
+        ":$TEXINPUTS:",
+    )
+    setup_env_for(type_env, "OPENTYPEFONTS", files)
+    setup_env_for(type_env, "T1FONTS", files, [".pfa", ".pfb"])
+    setup_env_for(type_env, "TEXFONTMAPS", files, [".map"], ":")
+    setup_env_for(type_env, "TEXINPUTS", files)
+    setup_env_for(type_env, "TFMFONTS", files, [".tfm"])
+    setup_env_for(type_env, "TTFONTS", files, [".ttf", ".ttc"])
+    setup_env_for(type_env, "VFFONTS", files, [".vf"])
+
+    env = {
+        "PATH": ":".join(
+            [
+                toolchain.kpsewhich.files.to_list()[0].dirname,  # latex bin folder
+                toolchain.mktexlsr.files.to_list()[0].dirname,  # script folder
+                "/bin",  # sed, rm, etc. needed by mktexlsr
+                "/usr/bin",  # needed to find python
```

Comment:
It should be possible to remove.
IIRC, this was only due latexrun being provided through exports_files.

I can change it py_binary instead

Reply:
changed to py_binary, yet, it fails, I wonder if setting the env parameter overwrites some default env values...

Reply:
This is related to:
https://github.com/bazelbuild/bazel/issues/4912
ctx.configuration.default_shell_env used to contain the default env such that it could be read and modified appended to. But is currently not.

I still think I prefer this over a wrapper. This should be ironed out any year now :P

Reply:
I guess, as you mention, if we define (maybe build) our own interpreter, that we won't need to depend on the host python interpreter.

But maybe that is for another pull-request

Reply:
Sounds good for now.